src/tests/thirdparty/t_parmetis_smoke.cpp

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) Lawrence Livermore National Security, LLC and other Conduit
+// Project developers. See top-level LICENSE AND COPYRIGHT files for dates and
+// other details. No copyright assignment is required to contribute to Conduit.
+
+//-----------------------------------------------------------------------------
+///
+/// file: t_parmetis_smoke.cpp
+///
+//-----------------------------------------------------------------------------
+
+#include <mpi.h>
+#include <iostream>
+#include <fstream>
+
+#include <parmetis.h>
+
+//------------------------------------------------------------------------------
+int main(int argc, char** argv)
+{
+    // Initialize MPI and get rank and comm size
+    MPI_Init(&argc, &argv);
+
+    int par_rank, par_size;
+    MPI_Comm_rank(MPI_COMM_WORLD, &par_rank);
+    MPI_Comm_size(MPI_COMM_WORLD, &par_size);
+
+    //
+    //   Simple example mesh
+    //
+    //   0 ---- 1 ---- 2
+    //   |  e0  |  e1  |
+    //   3 ---- 4 ---  5
+    //   |  e2  |  e3  |
+    //   6 ---- 7 ---- 8
+
+
+    // elems across processors,
+    // first rank has [0,1,2], second rank has [3]
+    idx_t eldist[] = {0, 3, 4};
+
+    /*
+    The eptr and eind arrays are similar in nature to the xadj and adjncy
+    arrays used to specify the adjacency list of a graph but now for each
+    element they specify the set of nodes that make up each element.
+    Specifically, the set of nodes that belong to element i is stored in
+    array eind starting at index eptr[i] and ending at (but not including)
+    index eptr[i + 1] (in other words, eind[eptr[i]] up through and including
+    eind[eptr[i + 1]-1]). Hence, the node lists for each element are stored
+    consecutively in the array eind.
```

Comment:
It looks like METIS uses the same offsetting mechanism as some of our client codes; interesting!